src/bin/pg_autoctl/fsm_transition.c

```diff
@@ -846,7 +898,8 @@ fsm_start_maintenance_on_standby(Keeper *keeper)
 bool
 fsm_restart_standby(Keeper *keeper)
 {
-	return fsm_start_postgres(keeper);
+	/* the primary might have change during our maintenance period */
+	return fsm_follow_new_primary(keeper);
```

Comment:
Meanwhile in the main branch the code for restarting a standby after maintenance is now:

``` c
bool
fsm_restart_standby(Keeper *keeper)
{
	return fsm_rewind_or_init(keeper);
}
```

And this will fetch the current primary from the monitor. I think this also is going to be changed in the big merge to the current main branch.